source/common/stats/symbol_table_impl.h

```diff
@@ -356,6 +422,18 @@ class StatNameJoiner {
   std::unique_ptr<SymbolStorage> bytes_;
 };
 
+class StatNameTempStorage : public StatNameStorage {
```

Comment:
Please document why this class exists and how it differs from StatNameStorage, what lifetime is implied by its name, etc.